lib/provisioner/worker.rb

```diff
@@ -0,0 +1,322 @@
+      clazz = nil # rubocop:disable UselessAssignment
+      object = nil
+      result = nil
+      classes = nil
+      task_id = task['taskId']
+
+      log.debug "Processing task with id #{task_id} ..."
+
+      taskName = task['taskName'].downcase
+      # depending on task, these may be nil
+      # automator take pecedence as presence indicates a 'software' task
+      providerName = task['config']['provider']['providertype'] rescue nil
+      automatorName = task['config']['service']['action']['type'] rescue nil
+
+      case taskName.downcase
+      when 'create', 'confirm', 'delete'
+        clazz = Object.const_get(pluginmanager.getHandlerActionObjectForProvider(providerName))
+        cwd = File.join(@config.get(PROVISIONER_WORK_DIR), @tenant, 'providertypes', providerName)
+        result = _run_plugin(clazz, @plugin_env, cwd, task)
+      when 'install', 'configure', 'initialize', 'start', 'stop', 'remove'
+        clazz = Object.const_get(pluginmanager.getHandlerActionObjectForAutomator(automatorName))
+        cwd = File.join(@config.get(PROVISIONER_WORK_DIR), @tenant, 'automatortypes', automatorName)
+        result = _run_plugin(clazz, @plugin_env, cwd, task)
+      when 'bootstrap'
+        combinedresult = {}
+        classes = []
+        if task['config'].key? 'automators' and !task['config']['automators'].empty?
+          # server must specify which bootstrap handlers need to run
+          log.debug "Task #{task_id} running specified bootstrap handlers: #{task['config']['automators']}"
+          task['config']['automators'].each do |automator|
+            clazz = Object.const_get(pluginmanager.getHandlerActionObjectForAutomator(automator))
+            cwd = File.join(@config.get(PROVISIONER_WORK_DIR), @tenant, 'automatortypes', automator)
+            result = _run_plugin(clazz, @plugin_env, cwd, task)
+            combinedresult.merge!(result)
+          end
+        else
+          log.warn 'No automators specified to bootstrap'
+        end
+        result = combinedresult
+      else
+        log.error "Unhandled task of type #{task['taskName']}"
+        fail "Unhandled task of type #{task['taskName']}"
+      end
+      result
+    end
+
+    # Run a single task read from file
+    def run_task_from_file
+      begin
+        result = nil
+        task = nil
+        log.info "Start Provisioner run for file #{@file}"
+        task = JSON.parse(IO.read(@file))
+
+        # While provisioning, don't allow the provisioner to terminate by disabling signal
+        sigterm = SignalHandler.new('TERM')
+        sigterm.dont_interupt {
+          result = delegate_task(task, @pluginmanager) # TODO: we dont need to pass pluginmanager anymore
+        }
+      rescue => e
+        log.error "Caught exception when running task from file #{@file}"
+
+        result = {} if result.nil? == true
+        result['status'] = '1'
+        if e.class.name == 'CommandExecutionError'
```

Comment:
What is the reason for this conditional? It's not entirely clear. A small comment would help.